pennylane/estimator/templates/qubitize.py

```diff
@@ -0,0 +1,369 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Resource operators for PennyLane subroutine templates."""
+import math
+
+import numpy as np
+
+from pennylane.estimator.compact_hamiltonian import THCHamiltonian
+from pennylane.estimator.ops.op_math.controlled_ops import MultiControlledX, Toffoli
+from pennylane.estimator.ops.op_math.symbolic import Adjoint, Controlled
+from pennylane.estimator.ops.qubit.non_parametric_ops import X
+from pennylane.estimator.resource_operator import (
+    CompressedResourceOp,
+    GateCount,
+    ResourceOperator,
+    resource_rep,
+)
+from pennylane.estimator.templates.select import SelectTHC
+from pennylane.estimator.templates.stateprep import PrepTHC
+from pennylane.estimator.wires_manager import Allocate, Deallocate
+from pennylane.wires import Wires, WiresLike
+
+# pylint: disable=signature-differs, arguments-differ, too-many-arguments
+
+
+class QubitizeTHC(ResourceOperator):
+    r"""Resource class for qubitization of tensor hypercontracted Hamiltonian.
+
+    .. note::
+
+            This decomposition assumes that an appropriately sized phase gradient state is available.
+            Users should ensure that the cost of constructing this state has been accounted for.
+            See also :class:`~.pennylane.estimator.templates.PhaseGradient`.
+
+    Args:
+        thc_ham (:class:`~.pennylane.estimator.compact_hamiltonian.THCHamiltonian`): A tensor hypercontracted
+            Hamiltonian for which the walk operator is being created.
+        prep_op (:class:`~.pennylane.estimator.resource_operator.ResourceOperator` | None): An optional
+            resource operator, corresponding to the prepare routine. If :code:`None`, the
+            default :class:`~.pennylane.estimator.templates.PrepTHC` will be used.
+        select_op (:class:`~.pennylane.estimator.resource_operator.ResourceOperator` | None): An optional
+            resource operator, corresponding to the select routine. If :code:`None`, the
+            default :class:`~.pennylane.estimator.templates.SelectTHC` will be used.
```

Comment:
PrepTHC and SelectTHC links also don't work. I think you need to provide the full path (e.g `~.pennylane.estimator.templates.stateprep.PrepTHC`)